File: algorithms/medium/removing_stars_from_a_string.dart
class Solution {
  String removeStars(String s) {
    List<String> ans = [];
    for (int i = 0; i < s.length; i++) {
      if (s[i] == "*") {
        ans.removeLast();
      } else {
        ans.add(s[i]);
      }
    }
    return ans.join("");
  }
}

File: algorithms/medium/removing_stars_from_a_string.ex
defmodule Solution do
  @spec remove_stars(s :: String.t()) :: String.t()
  def remove_stars(s) do
    s
    |> String.graphemes()
    |> Enum.reduce([], fn c, acc ->
      cond do
        c == "*" && acc != [] -> tl(acc)
        c == "*" -> acc
        true -> [c | acc]
      end
    end)
    |> Enum.join("")
    |> String.reverse()
  end
end
